packages/driver/src/cy/commands/xhr.coffee

```diff
@@ -22,6 +22,9 @@ reset = ->
 
   server = null
 
+  # TODO reset returns a promise, so all uses should be updated
+  Cypress.backend("set:traffic:routing:reset")
```

Comment:
Your reset function is sync, but needs to tap into test:before:run:async and use a callback to ensure that the server ack's the reset before each test